server/enhancedDate.js

```diff
@@ -1 +1,80 @@
-// export this as a Node module using the enhancedDate.js code from Module_2
\ No newline at end of file
+// export this as a Node module using the enhancedDate.js code from Module_2
+
+
+
+  var dayNames = ['Sunday','Monday','Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+  var monthNames = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+  var myDate=null;
+  var dateIsSet= false;
+  var myDayName;
+  var todaysDate;
+  var myMonth;
+    function setDate(date) {
+        if (date instanceof Date || typeof enterDate === 'number'){
+          myDate = new Date(date);
+
+        } else{
+          myDate = new Date();
+        }
+    }
+
+
+    function getDate(returnObject) {
+         if (!dateIsSet){
+         	setDate();
```

Comment:
Either setDate should set dateIsSet to true or you need to set it to true after calilng setDate, otherwise you're setting a new date each time you call a method on your module.